### ISSUE TITLE
Include metadata in scans as well as when called with the --metadata option.

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-linter#557]
Not sure how to call this but having the addon metadata alongside the validation result is helpful for showing more details and extracting `architecture`.

Maybe report `metadata` by default (and keep have `--metadata` to only see the metadata) or add a new `--full` option that merges both.

**Comment by muffinresearch** [comment]
Looking at the code the `extractMetadata` method is called when the metadata option is true *and* for a normal scan. Is it the case that you're not seeing the metadata output when you are scanning - if so that's definitely a bug.

**Comment by EnTeQuAk** [comment]
Okay, I think I confused my self and the issue seems to be something else. These are two runs, once with `--metadata` and once without:

```bash
$ addons-linter ./src/olympia/files/fixtures/files/webextension.xpi --pretty --metadata
{
    "hasErrors": true,
    "metadata": {
        "manifestVersion": 2,
        "name": "My WebExtension Addon",
        "type": 1,
        "version": "0.0.1",
        "architecture": "extension",
        "emptyFiles": [],
        "jsLibs": {}
    },
    "errors": [
        {
            "_type": "error",
            "code": "MANIFEST_JSON_INVALID",
            "message": "/applications/gecko/id: should match pattern \"^{[A-Fa-f0-9]{8}-([A-Fa-f0-9]{4}-){3}[A-Fa-f0-9]{12}}$|^[A-Za-z0-9-._]+\\@[A-Za-z0-9-._]+$\"",
            "description": "id is the extension ID. Mandatory. See https://developer.mozilla.org/Add-ons/Install_Manifests#id",
            "file": "manifest.json"
        }
    ]
}
```
```bash
$ addons-linter ./src/olympia/files/fixtures/files/webextension.xpi --pretty
{
    "count": 1,
    "summary": {
        "errors": 1,
        "notices": 0,
        "warnings": 0
    },
    "errors": [
        {
            "_type": "error",
            "code": "MANIFEST_JSON_INVALID",
            "message": "/applications/gecko/id: should match pattern \"^{[A-Fa-f0-9]{8}-([A-Fa-f0-9]{4}-){3}[A-Fa-f0-9]{12}}$|^[A-Za-z0-9-._]+\\@[A-Za-z0-9-._]+$\"",
            "description": "id is the extension ID. Mandatory. See https://developer.mozilla.org/Add-ons/Install_Manifests#id",
            "file": "manifest.json"
        }
    ],
    "notices": [],
    "warnings": []
}
```

So yeah, metadata is returned but the returned JSON has a totally different structure (no `"notices"`, no `"warnings"` and no `"summary"`).

Is that expected?

**Comment by muffinresearch** [comment]
Ok - the second one should include the metadata as well. So that's what we should fix.

**Comment by muffinresearch** [comment]
Looks like this if statement needs to be removed.

**Comment by EnTeQuAk** [comment]
yeah, that was the point I was trying to make :)
